packages/@aws-cdk/aws-lambda/lib/function-url.ts

```diff
@@ -0,0 +1,195 @@
+import { IResource, Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { IFunction } from './function-base';
+import { CfnUrl } from './lambda.generated';
+
+
+/**
+ * The auth types for a function url
+ */
+export enum FunctionUrlAuthType {
+  /**
+   * Restrict access to authenticated IAM users only
+   */
+  AWS_IAM = 'AWS_IAM',
+
+  /**
+   * Bypass IAM authentication to create a public endpoint
+   */
+  NONE = 'NONE',
+}
+
+/**
+ * All http request methods
+ */
+export enum HttpMethods {
+  /**
+   * The GET method requests a representation of the specified resource.
+   */
+  GET = 'GET',
+  /**
+   * The PUT method replaces all current representations of the target resource with the request payload.
+   */
+  PUT = 'PUT',
+  /**
+   * The HEAD method asks for a response identical to that of a GET request, but without the response body.
+   */
+  HEAD = 'HEAD',
+  /**
+   * The POST method is used to submit an entity to the specified resource, often causing a change in state or side effects on the server.
+   */
+  POST = 'POST',
+  /**
+   * The DELETE method deletes the specified resource.
+   */
+  DELETE = 'DELETE',
+}
+
+/**
+ * Specifies a cross-origin access property for a function URL
+ */
+export interface CorsProperty {
+  /**
+   * Whether to allow cookies or other credentials in requests to your function URL.
+   *
+   * @default false
+   */
+  readonly allowCredentials?: boolean;
+
+  /**
+   * Headers that are specified in the Access-Control-Request-Headers header.
+   *
+   * @default - No headers allowed.
+   */
+  readonly allowedHeaders?: string[];
+
+  /**
+   * An HTTP method that you allow the origin to execute.
+   *
+   * @default - No methods allowed.
+   */
+  readonly allowedMethods?: HttpMethods[];
+
+  /**
+   * One or more origins you want customers to be able to access the bucket from.
+   *
+   * @default - No origins allowed.
+   */
+  readonly allowedOrigins?: string[];
+
+  /**
+   * One or more headers in the response that you want customers to be able to access from their applications.
+   *
+   * @default - No headers exposed.
+   */
+  readonly exposedHeaders?: string[];
+
+  /**
+   * The time in seconds that your browser is to cache the preflight response for the specified resource.
+   *
+   * @default - No caching.
+   */
+  readonly maxAge?: number;
```

Comment:
This needs to b a `cdk.Duration`.

Reply:
We have followed similar type nomenclature at other places where we define `CORS` related config. Example S3: 
https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-s3/lib/bucket.ts#L1048

Reply:
That is old. Our guidance has evolved over time.